File: src/py-builder/smart-solution.py
# For users: 

# This is smart solution (uses greedy aproach).

# You should write code as usual(cin, cout),
# no need to work with any threads,
# all the hard work is done for you

import sys

INF = 10**9

def solve():
    data = sys.stdin.read().split()
    n, m = map(int, data[:2])
    a = list(map(int, data[2:2+n]))

    a.sort(reverse=True)

    cnt = 0
    remaining = m
    for x in a:
        while remaining - x >= 0:
            remaining -= x
            cnt += 1

    if remaining > 0:
        print(-1)
    else:
        print(cnt)

solve()

File: src/runners/go-runner/smart-solution.go
/*

For users:

This is smart solution (uses greedy aproach).

You should write code as usual(cin, cout),
no need to work with any threads,
all the hard work is done for you

*/

package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	var n, m int
	fmt.Fscan(in, &n, &m)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Fscan(in, &a[i])
	}

	sort.Slice(a, func(i, j int) bool { return a[i] > a[j] })

	ans := 0
	for i := 0; i < n; i++ {
		for m-a[i] >= 0 {
			m -= a[i]
			ans++
		}
	}

	if m > 0 {
		fmt.Println(-1)
	} else {
		fmt.Println(ans)
	}
}
